sigstore-java/src/main/java/dev/sigstore/rekor/client/RekorVerifier.java

```diff
@@ -92,49 +95,31 @@ public void verifyEntry(RekorEntry entry) throws RekorVerificationException {
 
   /** Verify that a Rekor Entry is in the log by checking inclusion proof. */
   private void verifyInclusionProof(RekorEntry entry) throws RekorVerificationException {
-
     var inclusionProof = entry.getVerification().getInclusionProof();
 
     var leafHash =
         Hashing.sha256()
-            .hashBytes(combineBytes(new byte[] {0x00}, Base64.getDecoder().decode(entry.getBody())))
+            .hashBytes(
+                InclusionProofVerifier.combineBytes(
+                    new byte[] {0x00}, Base64.getDecoder().decode(entry.getBody())))
             .asBytes();
 
-    // see: https://datatracker.ietf.org/doc/rfc9162/ section 2.1.3.2
-
-    // nodeIndex and totalNodes represent values for a specific level in the tree
-    // starting at the leafs and moving up to the root.
-    var nodeIndex = inclusionProof.getLogIndex();
-    var totalNodes = inclusionProof.getTreeSize() - 1;
-
-    var currentHash = leafHash;
-    var hashes = inclusionProof.getHashes();
-
-    for (var hash : hashes) {
-      byte[] p = Hex.decode(hash);
-      if (totalNodes == 0) {
-        throw new RekorVerificationException("Inclusion proof failed, ended prematurely");
-      }
-      if (nodeIndex == totalNodes || nodeIndex % 2 == 1) {
-        currentHash = hashChildren(p, currentHash);
-        while (nodeIndex % 2 == 0) {
-          nodeIndex = nodeIndex >> 1;
-          totalNodes = totalNodes >> 1;
-        }
-      } else {
-        currentHash = hashChildren(currentHash, p);
-      }
-      nodeIndex = nodeIndex >> 1;
-      totalNodes = totalNodes >> 1;
+    List<byte[]> hashes = new ArrayList<>();
+    for (String hash : inclusionProof.getHashes()) {
+      hashes.add(Hex.decode(hash));
```

Comment:
By the way, just to know for the future, in a TransparencyLogEntry, the hashes are not hex-encoded, that's only for RekorEntry.